src/components/interactions/GraspGauge.tsx

```diff
@@ -0,0 +1,79 @@
+import { useState } from "react";
+import { useSocketContext } from "../contexts/socket/useSocketContext";
+import ToggleButtonGroup from "@mui/material/ToggleButtonGroup";
+import ToggleButton from "@mui/material/ToggleButton";
+import { Box, Typography } from "@mui/material";
+import useThrottle from "@/hooks/useThrottle";
+
+interface GraspReaction {
+  level: string;
+  emoji: string;
+}
+
+const graspReactions: GraspReaction[] = [
+  { level: "Good", emoji: "😌" },
+  { level: "Unsure", emoji: "🤔" },
+  { level: "Lost", emoji: "😳" },
+];
+
+const GraspGauge = () => {
+  const socket = useSocketContext();
+  const [currentGrasp, setCurrentGrasp] = useState<GraspReaction | null>(null);
+  const [throttledValue, isDisabled] = useThrottle(currentGrasp, 3000);
+
+  const handleGraspChange = (
+    _event: React.MouseEvent<HTMLElement>,
+    updatedGrasp: GraspReaction | null,
+  ) => {
+    setCurrentGrasp(updatedGrasp);
+    if (socket && updatedGrasp) {
+      socket.emit(
+        "GraspReactionSend",
+        {
+          level: updatedGrasp.level.toLowerCase(),
+          sent_at: new Date(),
+        },
+        (success: boolean) => {
+          if (success) {
+            // Immediately reset selection to setup for new selection
+            setCurrentGrasp(null);
+          }
+        },
+      );
+    }
+  };
+
+  const handleGraspClick = (
+    event: React.MouseEvent<HTMLElement>,
+    codePoint?: number,
+  ) => {
+    if (socket && codePoint) {
+      socket.emit("ReactionSend", codePoint);
+    }
+  };
+
+  return (
+    <Box my={2}>
+      <Typography variant="h6">Grasp Gauge</Typography>
+      <ToggleButtonGroup
+        exclusive
+        disabled={isDisabled === true || false}
```

Comment:
When it's just `isDisabled` TypeScript errors with `Type 'null' is not assignable to type 'boolean | undefined'.ts(2322)`. 

I think because of the way `isDisabled` is initialized with `const [throttledValue, isDisabled] = useThrottle(currentGrasp, 3000);` on line 23.

Reply:
I think we might need to specify the return types for the custom hook explicitly. TS infers the return type as an array of elements, where each element has the union of all the possible types, e.g., `boolean | GraspReaction | null`. But in actuality the array always has length 2 and the second element is always a boolean and we want the types to reflect that information. I found the following [helpful](https://react-typescript-cheatsheet.netlify.app/docs/basic/getting-started/hooks/#custom-hooks) while just looking this up.